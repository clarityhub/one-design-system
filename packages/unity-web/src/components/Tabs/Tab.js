import React from 'react';
import { oneOfType, node, func, string } from 'prop-types';
import { ButtonGroupItem } from '../ButtonGroup';

const Tab = ({ children, tab, _onClick, _currentTab, ...props }) => {
	const handleClick = (e) => {
		_onClick(e, tab);
	};

	const selected = tab === _currentTab;

	return (
		<ButtonGroupItem {...props} selected={selected} onClick={handleClick}>
			{children}
		</ButtonGroupItem>
	);
};

Tab.propTypes = {
	children: oneOfType([node, func]).isRequired,
	tab: string.isRequired,
};

export default Tab;
