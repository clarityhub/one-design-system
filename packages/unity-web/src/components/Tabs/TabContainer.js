import React, { useState } from 'react';
import { oneOfType, node, func, string } from 'prop-types';

const getDefaultTabFromChildren = (children) => {
	try {
		const arr = React.Children.toArray(children);
		if (arr && arr.length > 0) {
			const tabs = React.Children.toArray(arr[0].props.children);

			if (tabs && tabs.length > 0) {
				return tabs[0].props.tab;
			}
		}
	} catch (e) {
		// Do nothing
	}

	return null;
};

const TabContainer = ({ children, defaultTab, onChange = () => {} }) => {
	const [currentTab, setCurrentTab] = useState(() => {
		if (!defaultTab) {
			return getDefaultTabFromChildren(children);
		}

		return defaultTab;
	});

	const handleClick = (e, tab) => {
		setCurrentTab(tab);
		onChange(tab);
	};

	return React.Children.map(children, (child) => {
		return React.cloneElement(child, {
			_currentTab: currentTab,
			_onClick: handleClick,
		});
	});
};

TabContainer.propTypes = {
	children: oneOfType([node, func]),
	defaultTab: string,
	onChange: func,
};

export default TabContainer;
