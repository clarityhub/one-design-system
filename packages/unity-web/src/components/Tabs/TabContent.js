import { string, oneOfType, node, func } from 'prop-types';

const TabContent = ({ for: propFor, _currentTab, children }) => {
	if (propFor === _currentTab) {
		return children;
	}

	return null;
};

TabContent.propTypes = {
	children: oneOfType([node, func]),
	for: string.isRequired,
};

export default TabContent;
