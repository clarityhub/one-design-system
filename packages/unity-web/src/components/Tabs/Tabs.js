import React from 'react';
import { oneOfType, node, func } from 'prop-types';
import { ButtonGroup } from '../ButtonGroup';

const Tabs = ({ children, _onClick, _currentTab, ...props }) => {
	return (
		<div {...props}>
			<ButtonGroup>
				{React.Children.map(children, (child) => {
					return React.cloneElement(child, {
						_onClick: _onClick,
						_currentTab: _currentTab,
					});
				})}
			</ButtonGroup>
		</div>
	);
};

Tabs.propTypes = {
	children: oneOfType([node, func]),
};

export default Tabs;
