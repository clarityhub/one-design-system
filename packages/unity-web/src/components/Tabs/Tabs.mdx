---
name: Tabs
menu: Components
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import Tabs from '.';
import TabContainer from './TabContainer';
import TabContent from './TabContent';

# Tabs

Use Tabs when you want to display different information depending on a selection.

```js
import Tabs from '@clarityhub/unity-web/lib/components/Tabs';
```

## Examples

<Playground>
    <Tabs.Container>
        <Tabs.Tabs>
            <Tabs.Tab tab="table" title="Switch to Table View">
                <span style={{ verticalAlign: 'middle' }}>
                    {' '}
                    Table
                </span>
            </Tabs.Tab>
            <Tabs.Tab tab="chart" title="Switch to Chart View">
                <span style={{ verticalAlign: 'middle' }}>
                    {' '}
                    Charts
                </span>
            </Tabs.Tab>
        </Tabs.Tabs>
        <Tabs.TabContent for="table">
            Table content
        </Tabs.TabContent>
        <Tabs.TabContent for="chart">
            Chart content
        </Tabs.TabContent>
    </Tabs.Container>
</Playground>

## Prop Types

### Tabs.Container
<PropsTable of={TabContainer} />

### Tabs.Tabs
<PropsTable of={Tabs.Tabs} />

### Tabs.Tab
<PropsTable of={Tabs.Tab} />

### Tabs.TabContent
<PropsTable of={TabContent} />

