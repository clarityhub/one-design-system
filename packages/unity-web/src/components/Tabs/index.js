import Tab from './Tab';
import TabContainer from './TabContainer';
import TabContent from './TabContent';
import Tabs from './Tabs';

const TabGroup = {};
TabGroup.Container = TabContainer;
TabGroup.TabContent = TabContent;
TabGroup.Tab = Tab;
TabGroup.Tabs = Tabs;

export default TabGroup;
