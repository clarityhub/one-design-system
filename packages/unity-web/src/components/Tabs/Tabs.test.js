import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Tabs from './';

describe('<Tabs />', () => {
	it('renders a set of tabs', () => {
		const node = (
			<Tabs.Container>
				<Tabs.Tabs>
					<Tabs.Tab tab="table" title="Switch to Table View">
						<span style={{ verticalAlign: 'middle' }}>
							{' '}
                            Table
						</span>
					</Tabs.Tab>
					<Tabs.Tab tab="chart" title="Switch to Chart View">
						<span style={{ verticalAlign: 'middle' }}>
							{' '}
                            Charts
						</span>
					</Tabs.Tab>
				</Tabs.Tabs>
				<Tabs.TabContent for="table">
                    Table content
				</Tabs.TabContent>
				<Tabs.TabContent for="chart">
                    Chart content
				</Tabs.TabContent>
			</Tabs.Container>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
    
	it('renders a different tab after click', () => {
		const node = (
			<Tabs.Container>
				<Tabs.Tabs>
					<Tabs.Tab tab="table" title="Switch to Table View">
						<span style={{ verticalAlign: 'middle' }}>
							{' '}
                            Table
						</span>
					</Tabs.Tab>
					<Tabs.Tab tab="chart" title="Switch to Chart View">
						<span style={{ verticalAlign: 'middle' }}>
							{' '}
                            Charts
						</span>
					</Tabs.Tab>
				</Tabs.Tabs>
				<Tabs.TabContent for="table">
                    Table content
				</Tabs.TabContent>
				<Tabs.TabContent for="chart">
                    Chart content
				</Tabs.TabContent>
			</Tabs.Container>
		);

		const component = renderer.create(node);

		act(component.root.findAllByType('button')[1].props.onClick);

		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});