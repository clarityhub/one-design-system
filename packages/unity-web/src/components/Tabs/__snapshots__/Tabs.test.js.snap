// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> renders a different tab after click 1`] = `
Array [
  .emotion-4 {
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  display: inline-block;
}

.emotion-4 .emotion-1 {
  box-shadow: none;
}

.emotion-4 .emotion-1 + .emotion-1 {
  border-left-width: 0;
  border-left: 1px solid #F1F1F1;
}

.emotion-4 .emotion-1:not(:last-child):not(:first-of-type) {
  border-radius: 0;
}

.emotion-4 .emotion-1:first-of-type {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-4 .emotion-1:last-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-0 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus,
.emotion-0.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-0:disabled,
.emotion-0.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-0 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-0.-default,
.emotion-0.-primary,
.emotion-0.-danger,
.emotion-0.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-0.-default.-outline,
.emotion-0.-primary.-outline,
.emotion-0.-danger.-outline,
.emotion-0.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-0.-default.-text,
.emotion-0.-primary.-text,
.emotion-0.-danger.-text,
.emotion-0.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-0.-default:hover,
.emotion-0.-primary:hover,
.emotion-0.-danger:hover,
.emotion-0.-white:hover,
.emotion-0.-default.active,
.emotion-0.-primary.active,
.emotion-0.-danger.active,
.emotion-0.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-0.-default:hover.-outline,
.emotion-0.-primary:hover.-outline,
.emotion-0.-danger:hover.-outline,
.emotion-0.-white:hover.-outline,
.emotion-0.-default.active.-outline,
.emotion-0.-primary.active.-outline,
.emotion-0.-danger.active.-outline,
.emotion-0.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-0.-default:hover.-text,
.emotion-0.-primary:hover.-text,
.emotion-0.-danger:hover.-text,
.emotion-0.-white:hover.-text,
.emotion-0.-default.active.-text,
.emotion-0.-primary.active.-text,
.emotion-0.-danger.active.-text,
.emotion-0.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-0.-default:disabled,
.emotion-0.-primary:disabled,
.emotion-0.-danger:disabled,
.emotion-0.-white:disabled,
.emotion-0.-default.disabled,
.emotion-0.-primary.disabled,
.emotion-0.-danger.disabled,
.emotion-0.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-0.-default:disabled.-outline,
.emotion-0.-primary:disabled.-outline,
.emotion-0.-danger:disabled.-outline,
.emotion-0.-white:disabled.-outline,
.emotion-0.-default.disabled.-outline,
.emotion-0.-primary.disabled.-outline,
.emotion-0.-danger.disabled.-outline,
.emotion-0.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-0.-default:disabled.-text,
.emotion-0.-primary:disabled.-text,
.emotion-0.-danger:disabled.-text,
.emotion-0.-white:disabled.-text,
.emotion-0.-default.disabled.-text,
.emotion-0.-primary.disabled.-text,
.emotion-0.-danger.disabled.-text,
.emotion-0.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-0.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-0.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-0.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-0.-text:hover,
.emotion-0.-text.active {
  border-color: transparent;
}

.emotion-0.-text.disabled,
.emotion-0.-text:disabled {
  border: 1px solid transparent;
}

.emotion-0.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-0.disabled,
.emotion-0:disabled {
  cursor: not-allowed;
}

<div>
    <div
      className="emotion-4 emotion-5"
    >
      <button
        className="-default emotion-0 emotion-1"
        onClick={[Function]}
        title="Switch to Table View"
        type="button"
      >
        <span
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
           
          Table
        </span>
      </button>
      <button
        className="-primary emotion-0 emotion-1"
        onClick={[Function]}
        title="Switch to Chart View"
        type="button"
      >
        <span
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
           
          Charts
        </span>
      </button>
    </div>
  </div>,
  "Chart content",
]
`;

exports[`<Tabs /> renders a set of tabs 1`] = `
Array [
  .emotion-4 {
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  display: inline-block;
}

.emotion-4 .emotion-1 {
  box-shadow: none;
}

.emotion-4 .emotion-1 + .emotion-1 {
  border-left-width: 0;
  border-left: 1px solid #F1F1F1;
}

.emotion-4 .emotion-1:not(:last-child):not(:first-of-type) {
  border-radius: 0;
}

.emotion-4 .emotion-1:first-of-type {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-4 .emotion-1:last-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-0 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus,
.emotion-0.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-0:disabled,
.emotion-0.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-0 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-0.-default,
.emotion-0.-primary,
.emotion-0.-danger,
.emotion-0.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-0.-default.-outline,
.emotion-0.-primary.-outline,
.emotion-0.-danger.-outline,
.emotion-0.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-0.-default.-text,
.emotion-0.-primary.-text,
.emotion-0.-danger.-text,
.emotion-0.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-0.-default:hover,
.emotion-0.-primary:hover,
.emotion-0.-danger:hover,
.emotion-0.-white:hover,
.emotion-0.-default.active,
.emotion-0.-primary.active,
.emotion-0.-danger.active,
.emotion-0.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-0.-default:hover.-outline,
.emotion-0.-primary:hover.-outline,
.emotion-0.-danger:hover.-outline,
.emotion-0.-white:hover.-outline,
.emotion-0.-default.active.-outline,
.emotion-0.-primary.active.-outline,
.emotion-0.-danger.active.-outline,
.emotion-0.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-0.-default:hover.-text,
.emotion-0.-primary:hover.-text,
.emotion-0.-danger:hover.-text,
.emotion-0.-white:hover.-text,
.emotion-0.-default.active.-text,
.emotion-0.-primary.active.-text,
.emotion-0.-danger.active.-text,
.emotion-0.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-0.-default:disabled,
.emotion-0.-primary:disabled,
.emotion-0.-danger:disabled,
.emotion-0.-white:disabled,
.emotion-0.-default.disabled,
.emotion-0.-primary.disabled,
.emotion-0.-danger.disabled,
.emotion-0.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-0.-default:disabled.-outline,
.emotion-0.-primary:disabled.-outline,
.emotion-0.-danger:disabled.-outline,
.emotion-0.-white:disabled.-outline,
.emotion-0.-default.disabled.-outline,
.emotion-0.-primary.disabled.-outline,
.emotion-0.-danger.disabled.-outline,
.emotion-0.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-0.-default:disabled.-text,
.emotion-0.-primary:disabled.-text,
.emotion-0.-danger:disabled.-text,
.emotion-0.-white:disabled.-text,
.emotion-0.-default.disabled.-text,
.emotion-0.-primary.disabled.-text,
.emotion-0.-danger.disabled.-text,
.emotion-0.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-0.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-0.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-0.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-0.-text:hover,
.emotion-0.-text.active {
  border-color: transparent;
}

.emotion-0.-text.disabled,
.emotion-0.-text:disabled {
  border: 1px solid transparent;
}

.emotion-0.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-0.disabled,
.emotion-0:disabled {
  cursor: not-allowed;
}

<div>
    <div
      className="emotion-4 emotion-5"
    >
      <button
        className="-primary emotion-0 emotion-1"
        onClick={[Function]}
        title="Switch to Table View"
        type="button"
      >
        <span
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
           
          Table
        </span>
      </button>
      <button
        className="-default emotion-0 emotion-1"
        onClick={[Function]}
        title="Switch to Chart View"
        type="button"
      >
        <span
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
           
          Charts
        </span>
      </button>
    </div>
  </div>,
  "Table content",
]
`;
